compiler/optimizer/RegisterCandidate.hpp

```diff
@@ -54,32 +54,32 @@ class GlobalSet
 public:
    GlobalSet(TR::Compilation * comp, TR::Region &region);
 
-   TR_BitVector * operator[](TR::Block *BB)
+   TR_BitVector *operator[](uint32_t symRefNum)
       {
-      uint32_t blockNum = BB->getNumber();
-      if (blockNum == TR::CFG::StartBlock || blockNum == TR::CFG::EndBlock)
-         return NULL;
-   
-      auto lookup = _refAutosPerBlock.find(blockNum);
-      if (lookup != _refAutosPerBlock.end())
+      if (!_collected)
+         collectBlocks();
+
+      auto lookup = _blocksPerAuto.find(symRefNum);
+      if (lookup != _blocksPerAuto.end())
          return lookup->second; 
-      return collectReferencedAutoSymRefs(BB);
+      return &_empty;
       }
 
-   bool isEmpty() { return _refAutosPerBlock.empty(); }
-   void makeEmpty() { _refAutosPerBlock.clear(); }
+   bool isEmpty() { return _blocksPerAuto.empty(); }
+   void makeEmpty() { _collected = false; _blocksPerAuto.clear(); }
 
 private:
 
-   TR_BitVector *collectReferencedAutoSymRefs(TR::Block * BB);
-   void collectReferencedAutoSymRefs(TR::Node * node, TR_BitVector * referencedAutos, TR::NodeChecklist &visited);
+   void collectBlocks();
+   void collectReferencedAutoSymRefs(TR::Node *node, TR_BitVector &referencedAutoSymRefs, TR::NodeChecklist &visited);
 
    TR::Region &_region;
-   TR_BitVector _empty;
    typedef TR::typed_allocator<std::pair<uint32_t, TR_BitVector*>, TR::Region&> RefMapAllocator;
    typedef std::less<uint32_t> RefMapComparator;
    typedef std::map<uint32_t, TR_BitVector*, RefMapComparator, RefMapAllocator> RefMap;
-   RefMap _refAutosPerBlock;
+   RefMap _blocksPerAuto;
+   TR_BitVector _empty;
```

Comment:
if this a singleton for representing empty then I think the name should be _EMPTY which is a hint at the singleton-ness